server/jetstream_batching.go

```diff
@@ -164,8 +164,8 @@ func checkMsgHeadersPreClusteredProposal(
 			fields := strings.Split(string(srchdr), " ")
 			origStream := fields[0]
 			origSubj := subject
-			if len(fields) >= 3 {
-				origSubj = fields[2]
+			if len(fields) >= 5 {
+				origSubj = fields[4]
```

Comment:
Should this remain compatible for non-2.12 source strings? If not longer or equal to 5, but longer than equal to 3, we would still use index 2?